tests/math/test_entropies_math.py

```diff
@@ -224,3 +224,154 @@ def test_size_mismatch(self, state0, state1):
+
+        assert np.allclose(grad, expected, rtol=1e-06, atol=1e-07)
+
+    @pytest.mark.parametrize("interface", "jax")
+    @pytest.mark.parametrize(
+        "state, expected",
+        [
+            ([1, 0], -np.log(1)),
+            ([0, 1], -np.log(1)),
+            ([1, 1] / np.sqrt(2), -1 / (1 * np.log(base))),
+            ([1, -1] / np.sqrt(2), -1 / (1 * np.log(base))),
+            (
+                [1 / np.sqrt(3), np.sqrt(2 / 3)],
+                (-(np.sqrt(56249041) + 7500) / 15000) / (1 * np.log(base)),
+            ),
+        ],
+    )
+    @pytest.mark.parametrize("base", base)
+    @pytest.mark.parametrize("check_state", check_states)
+    def test_grad(self, interface, state, expected, base, check_state):
+        """Test that the gradient of min_entropy works
+        with the JAX interface"""
+        state = qml.math.asarray(state, like=interface)
+        grad = jax.grad(qml.math.min_entropy)(state, base=base, check_state=check_state)
+
+        assert np.allclose(grad, expected, rtol=1e-06, atol=1e-07)
+
+    @pytest.mark.parametrize("interface", "jax")
+    @pytest.mark.parametrize(
+        "state, expected",
+        [
+            ([1, 0], -np.log(1)),
+            ([0, 1], -np.log(1)),
+            ([1, 1] / np.sqrt(2), -1 / (1 * np.log(base))),
+            ([1, -1] / np.sqrt(2), -1 / (1 * np.log(base))),
+            (
+                [1 / np.sqrt(3), np.sqrt(2 / 3)],
+                (-(np.sqrt(56249041) + 7500) / 15000) / (1 * np.log(base)),
+            ),
+        ],
+    )
+    @pytest.mark.parametrize("base", base)
+    @pytest.mark.parametrize("check_state", check_states)
+    def test_grad(self, interface, state, expected, base, check_state):
+        """Test that the gradient of min_entropy works
+        with the JAX-jit interface"""
+        state = qml.math.asarray(state, like=interface)
+        grad = jax.jit(jax.grad(qml.math.min_entropy))(state, base=base, check_state=check_state)
+
+        assert np.allclose(grad, expected, rtol=1e-06, atol=1e-07)
+
+    @pytest.mark.parametrize("interface", "tensorflow")
+    @pytest.mark.parametrize(
+        "state, expected",
+        [
+            ([1, 0], -np.log(1)),
+            ([0, 1], -np.log(1)),
+            ([1, 1] / np.sqrt(2), -1 / (1 * np.log(base))),
+            ([1, -1] / np.sqrt(2), -1 / (1 * np.log(base))),
+            (
+                [1 / np.sqrt(3), np.sqrt(2 / 3)],
+                (-(np.sqrt(56249041) + 7500) / 15000) / (1 * np.log(base)),
+            ),
+        ],
+    )
+    @pytest.mark.parametrize("base", base)
+    @pytest.mark.parametrize("check_state", check_states)
+    def test_grad(self, interface, state, expected, base, check_state):
+        """Test that the gradient of min_entropy works
+        with the tensorflow interface"""
+        state = qml.math.asarray(state, like=interface)
+        grad = tf.gradient(qml.math.min_entropy)(state, base=base, check_state=check_state)
```

Comment:
I don't think this is a valid way of computing gradients in TensorFlow. Check [here](https://github.com/PennyLaneAI/pennylane/blob/master/tests/qinfo/test_entropies.py#L205) for an example. Again, just replace the QNode with the `min_entropy` function directly.